### PR TITLE
update to openms 3.2.0

### PR DESCRIPTION
**Pull request by ypriverol**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/quantms/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/quantms _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

___

### **PR Type**
enhancement, dependencies

___

### **Description**
- Updated GitHub Actions `upload-artifact` version from v1 to v4, and modified artifact names to include GitHub run ID.
- Upgraded OpenMS third-party conda package version from 3.1.0 to 3.2.0 across multiple modules.
- Upgraded quantms-utils conda package version from 0.0.10 to 0.0.11 across multiple modules.
- Updated `nextflow.config` version from 1.3.0dev to 1.3.1dev.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>ci.yml</strong><dd><code>Update GitHub Actions artifact version and naming</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.github/workflows/ci.yml

<li>Updated GitHub Actions <code>upload-artifact</code> version from v1 to v4.<br> <li> Modified artifact names to include GitHub run ID.<br>

</details>

  </td>
  <td><a href="https://github.com/bigbio/quantms/pull/423/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03f">+6/-6</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Dependencies</strong></td><td><details><summary>14 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>dev.config</strong><dd><code>Upgrade OpenMS third-party package version</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

conf/dev.config

<li>Updated OpenMS third-party conda package version from 3.1.0 to 3.2.0.<br>

</details>

  </td>
  <td><a href="https://github.com/bigbio/quantms/pull/423/files#diff-47565a38356978e76966195aa56162ca90cb53bac9c19caf2c23cf1a0262c118">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.nf</strong><dd><code>Upgrade quantms-utils package version</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

modules/local/add_sage_feat/main.nf

- Updated quantms-utils conda package version from 0.0.10 to 0.0.11.

</details>

  </td>
  <td><a href="https://github.com/bigbio/quantms/pull/423/files#diff-ebefd4b5549bb17371aeae20850f64231ff5f7757e244f3d6e5dcff3863d4251">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.nf</strong><dd><code>Upgrade quantms-utils package version</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

modules/local/diannconvert/main.nf

- Updated quantms-utils conda package version from 0.0.10 to 0.0.11.

</details>

  </td>
  <td><a href="https://github.com/bigbio/quantms/pull/423/files#diff-82309b6a96468c517d4180097cd07494b49b1135d98940922af45e371f9d011c">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.nf</strong><dd><code>Upgrade quantms-utils package version</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

modules/local/extract_psm/main.nf

- Updated quantms-utils conda package version from 0.0.10 to 0.0.11.

</details>

  </td>
  <td><a href="https://github.com/bigbio/quantms/pull/423/files#diff-5e4b89feb19467bd87f0ee82a9e92ec9c59221e88f80e0513515fe025ffbb9d9">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.nf</strong><dd><code>Upgrade quantms-utils package version</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

modules/local/extract_sample/main.nf

- Updated quantms-utils conda package version from 0.0.10 to 0.0.11.

</details>

  </td>
  <td><a href="https://github.com/bigbio/quantms/pull/423/files#diff-5cadb1b2914479fabd47ef59344fbfd23cc2b305dd3383613005d8ec81731de4">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.nf</strong><dd><code>Upgrade quantms-utils package version</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

modules/local/generate_diann_cfg/main.nf

- Updated quantms-utils conda package version from 0.0.10 to 0.0.11.

</details>

  </td>
  <td><a href="https://github.com/bigbio/quantms/pull/423/files#diff-c5ac97ba549a9d66bdc08dffb05ba514a147028c17f835b0eeaa999bc7d2f8fa">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.nf</strong><dd><code>Upgrade quantms-utils package version</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

modules/local/ms2rescore/main.nf

- Updated quantms-utils conda package version from 0.0.10 to 0.0.11.

</details>

  </td>
  <td><a href="https://github.com/bigbio/quantms/pull/423/files#diff-cd242368949fc1dd58f4c79858bb5a2439b2f4d6c9eb3190f07c62707efb410f">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.nf</strong><dd><code>Upgrade quantms-utils package version</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

modules/local/mzmlstatistics/main.nf

- Updated quantms-utils conda package version from 0.0.10 to 0.0.11.

</details>

  </td>
  <td><a href="https://github.com/bigbio/quantms/pull/423/files#diff-eb109b74c19adaeb41f929c16e391dde4ff2b30c5c7bb253d4f05821650fbed5">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.nf</strong><dd><code>Upgrade OpenMS third-party package version</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

modules/local/openms/consensusid/main.nf

<li>Updated OpenMS third-party conda package version from 3.1.0 to 3.2.0.<br>

</details>

  </td>
  <td><a href="https://github.com/bigbio/quantms/pull/423/files#diff-39d01a458c9dcee44b25e063031e28203fc10d6eb06e8db2a6ec4fc199dedaf6">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.nf</strong><dd><code>Upgrade OpenMS third-party package version</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

modules/local/openms/decoydatabase/main.nf

<li>Updated OpenMS third-party conda package version from 3.1.0 to 3.2.0.<br>

</details>

  </td>
  <td><a href="https://github.com/bigbio/quantms/pull/423/files#diff-b250c482db2ebc87d150b774e2963a04b11771a99de423d2f7076425b9606202">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.nf</strong><dd><code>Upgrade OpenMS third-party package version</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

modules/local/openms/epifany/main.nf

<li>Updated OpenMS third-party conda package version from 3.1.0 to 3.2.0.<br>

</details>

  </td>
  <td><a href="https://github.com/bigbio/quantms/pull/423/files#diff-0af0ce8dea671b29879d483a4cbc1f0b02a11dce52c79cda9e2ce42dae0a7e3e">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.nf</strong><dd><code>Upgrade OpenMS third-party package version</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

modules/local/openms/extractpsmfeatures/main.nf

<li>Updated OpenMS third-party conda package version from 3.1.0 to 3.2.0.<br>

</details>

  </td>
  <td><a href="https://github.com/bigbio/quantms/pull/423/files#diff-b81bee562e55c4210f61cbcfbbe50acbc18b46c3d22c0525e8e13371acb53987">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.nf</strong><dd><code>Upgrade OpenMS third-party package version</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

modules/local/openms/falsediscoveryrate/main.nf

<li>Updated OpenMS third-party conda package version from 3.1.0 to 3.2.0.<br>

</details>

  </td>
  <td><a href="https://github.com/bigbio/quantms/pull/423/files#diff-dec881706c4fbf21f398b8adc2453c97492eb8dc6d43a6b46f072ea66890761b">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.nf</strong><dd><code>Upgrade OpenMS third-party package version</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

modules/local/openms/filemerge/main.nf

<li>Updated OpenMS third-party conda package version from 3.1.0 to 3.2.0.<br>

</details>

  </td>
  <td><a href="https://github.com/bigbio/quantms/pull/423/files#diff-78415c287afadf0fceaf89be972aa3aeed12e0858a94f8661df335373c923afc">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (jpfeuffer) [inline .github/workflows/ci.yml:1]: Why are the artifact names not unique?
They should be. Or at least should be possible to be made unique.

Comment (ypriverol) [reply .github/workflows/ci.yml:1]: if you see all of them are but for some reason they get overwrite `${{ env.TEST_PROFILE }}_${{ env.EXEC_PROFILE }}_${{ github.run_id }}_results`

Comment (ypriverol) [reply .github/workflows/ci.yml:1]:
```
      - uses: actions/upload-artifact@v4
        if: failure() || cancelled()
        name: Upload failed logs
        with:
          name: failed_logs_${{ env.TEST_PROFILE }}_${{ env.EXEC_PROFILE }}_${{ github.run_id }}
          path: failed_logs
          overwrite: true
      - uses: actions/upload-artifact@v4
        if: always()
        name: Upload results
        with:
          name: ${{ env.TEST_PROFILE }}_${{ env.EXEC_PROFILE }}_${{ github.run_id }}_results
          path: ${{ env.TEST_PROFILE }}_${{ env.EXEC_PROFILE }}_results
          overwrite: true
      - uses: actions/upload-artifact@v4
        if: always()
        name: Upload log
        with:
          name: nextflow_${{ env.TEST_PROFILE }}_${{ env.EXEC_PROFILE }}_${{ github.run_id }}.log
          path: .nextflow.log
          overwrite: true
```

**Comment by jpfeuffer** [comment]
@timosachsenberg PR https://github.com/OpenMS/OpenMS/pull/7326 apparently broke the workflow. Any idea?
I checked. We are not overwriting any correction factor. Defaults should be used. Apparently they do not work anymore.

**Comment by timosachsenberg** [comment]
@cbielow you have a deeper understanding of that code. Any idea?

**Comment by jpfeuffer** [comment]
I am guessing that isobaricanalyzer is broken in general and only works in our tests because we completely supply an ini file with prefilled correction matrix.
Probably taking the default is not working.

**Comment by ypriverol** [comment]
@jpfeuffer @timosachsenberg @cbielow Taking the default will not work because that option do not exist anymore by reading this https://github.com/OpenMS/OpenMS/blob/ff325fe8572c953a6dbad3eb14b7facebf5549e5/src/topp/IsobaricAnalyzer.cpp#L95 looks like we have to provide for every method the matrix (then the concept of default is gone). Where and how I can generate what it used to be default metrics/values?

**Comment by timosachsenberg** [comment]
- @ypriverol do we specify that we want to do tmt11plex to IsobaricAnalyzer (e.g., given by the sdrf?)
- in IsobaricAnalyzer the tmt11plex has a zero correction matrix. Does anyone know if this is correct? I would assume that this leads to an identity matrix and the mentioned error. If it is correct, then an identity matrix would indeed be fine and correction can be skipped.

image

**Comment by timosachsenberg** [comment]
support was added in: https://github.com/OpenMS/OpenMS/commit/f4cee4e78e450d973bb3e765fa73b42e773a2b05

**Comment by timosachsenberg** [comment]
If I look at some batch **certificates** e.g. A37725_ZF395505 it seems as if tmt11plex is similar to tmt10plex and not all zeros?
https://www.thermofisher.com/order/catalog/product/de/en/A34808
So my guess would be that the entries are not correct in OpenMS?
Can someone confirm (@cbielow or @jpfeuffer)

**Comment by ypriverol** [comment]
> * @ypriverol do we specify that we want to do tmt11plex to IsobaricAnalyzer (e.g., given by the sdrf?)
> * in IsobaricAnalyzer the tmt11plex has a zero correction matrix. Does anyone know if this is correct? I would assume that this leads to an identity matrix and the mentioned error. If it is correct, then an identity matrix would indeed be fine and correction can be skipped.
> 
> image

@timosachsenberg if you see how we call the method: 
```
IsobaricAnalyzer \
      -type tmt6plex \
      -in TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01.mzML \
      -threads 2 \
      -extraction:select_activation "auto" \
      -extraction:reporter_mass_shift 0.002 \
      -extraction:min_reporter_intensity 0.0 \
      -extraction:min_precursor_purity 0.0 \
      -extraction:precursor_isotope_deviation 10.0 \
       \
      -tmt6plex:reference_channel 126 \
      -out TMT_Erwinia_1uLSike_Top10HCD_isol2_45stepped_60min_01_iso.consensusXML \
      -debug 0 \
      2>&1 | tee TMT_Erwinia_1uLSike_Top10HCD_isol2_[45](https://github.com/bigbio/quantms/actions/runs/11147642102/job/30982471524?pr=423#step:7:46)stepped_60min_01_isob.log
  
  cat <<-END_VERSIONS > versions.yml
  "NFCORE_QUANTMS:QUANTMS:TMT:FEATUREMAPPER:ISOBARICANALYZER":
      IsobaricAnalyzer: $(IsobaricAnalyzer --version 2>&1 | grep -E '^Version(.*)' | sed 's/Version: //g' | cut -d ' ' -f 1)
  END_VERSIONS
```
We do provide from the sdrf the type of tmt experiment in the current failing github actions is tmt6plex.

**Comment by timosachsenberg** [comment]
Seems it also has all zeros for TMT6Plex:
https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/ANALYSIS/QUANTITATION/TMTSixPlexQuantitationMethod.cpp#L46-L58

**Comment by timosachsenberg** [comment]
- @yasset as a temporary fix we need to specify the correction matrix in the ini file (to overwrite the correction matrix entries with zeros)
- @jpfeuffer and @cbielow @poshul correct me if I am wrong, but I think we also need to update those correction matrix entries.
 For tmt11plex see discussion: https://github.com/OpenMS/OpenMS/pull/4295#discussion_r337539997 why matrices were not averaged and we ended up with the default (zeros).

I see two possibilities to fix this in OpenMS:
- allow zeros again
- come up with a proper (averaged) matrix for the ones that are still zero

**Comment by jpfeuffer** [comment]
Then the problem is that we only make an exception for tmt11plex in the tool check. But not for tmt6plex which is apparently also all 0s.
I think we should definitely provide reasonable defaults, since it is very tedious (also just for testing) to specify long lists of strings on the command line.

**Comment by jpfeuffer** [comment]
My vote would be:
- If there is a clear default for a kit use it (but allow overwrite)
- If there is a reasonable default (most of the kits have that or if we can do some average), put it as default but warn
- If there is not a reasonable default put 0s, allow them (either per default or maybe only through a force flag)

**Comment by ypriverol** [comment]
I am fine with having default tables that we can provide to the tool. But we have to know where they are and how they should be annotated. I can store them in quantms as additional data. Then, if the users have a different correction matrix they should be able to provide it. See also this discussion in SDRF about TMT matrices standardisation: https://github.com/bigbio/proteomics-sample-metadata/issues/723 I can do that right now, but I need the "default" matrices for our previous implementation.

**Comment by jpfeuffer** [comment]
They are in the TMT*QuantitationMethod.cpp files that Timo linked. The problem is we do not have them for every TMT type. You have to come up with a good default yourself (based on the values on the vendor's Kit sheet, see e.g. Thermo's website).
Sometimes you get a specific matrix when buying the kit (based on the batch).

**Comment by ypriverol** [comment]
Do you know what others uses as default MaxQuant and others?

**Comment by jpfeuffer** [comment]
No. I would guess identity matrix or fail per default (i.e. matrix is mandatory)

**Comment by ypriverol** [comment]
@timosachsenberg @jpfeuffer I found the response about FragPipe https://github.com/Nesvilab/FragPipe/issues/963

**Comment by timosachsenberg** [comment]
The algorithm has an option enable_isotopic_correction or similar. Maybe it can be disabled in the tool via command line option. On my phone right now…

File: modules/local/openms/isobaricanalyzer/main.nf [hunk 1]
```diff
@@ -39,6 +39,7 @@ process ISOBARICANALYZER {
         -extraction:precursor_isotope_deviation $params.precursor_isotope_deviation \\
         ${iso_normalization} \\
         -${meta.labelling_type}:reference_channel $params.reference_channel \\
+        --quantificazion:isotope_correction false \\
```
Comment (timosachsenberg) [inline modules/local/openms/isobaricanalyzer/main.nf:42]: Typo?

**Comment by ypriverol** [comment]
@timosachsenberg @jpfeuffer Looks like it works?

**Comment by ypriverol** [comment]
@cbielow Thanks for the clarification. For now I'm first switching off (by default) and currently implementing a mechanism to provide a default matrix. BTW, Would be great if we can put somewhere (all the matrices) (not inside the code) to dynamically loaded by the user. Also for the user to know how to overwrite a matrix.

**Comment by ypriverol** [comment]
@timosachsenberg here is the first prototype about how the matrix should look like https://github.com/bigbio/quantms/blob/f1210fc56359f0e5f32268aa189f84d0b8a7000f/assets/tmt6plex_default_correction.matrix

**Comment by timosachsenberg** [comment]
I added one from existing lots: https://github.com/OpenMS/OpenMS/pull/7601
but one can of course just disable isotope correction. then it will be equivalent to your default one.
I did not implement loading from files in OpenMS but I think this is probably easier done with a script like yours that changes the ini.

**Comment by ypriverol** [comment]
I think for quantms it does't make sense now to have it in OpenMS, I will start adding some of the default matrices in assets and the users can select one of them if they want.

File: .github/workflows/ci.yml
Comment (jpfeuffer) [reply .github/workflows/ci.yml:1]: I am not sure if runid is unique for every matrix run. I think there is one ID for the whole workflow run.

This makes things like lfq and lfq_sage non-unique

Comment (ypriverol) [reply .github/workflows/ci.yml:1]:
```
      - uses: actions/upload-artifact@v4
        if: failure() || cancelled()
        name: Upload failed logs
        with:
          name: failed_logs_${{ matrix.test_profile }}_${{ matrix.exec_profile }}_${{ github.run_id }}_${{ github.job }}
          path: failed_logs
          overwrite: true

      - uses: actions/upload-artifact@v4
        if: always()
        name: Upload results
        with:
          name: ${{ matrix.test_profile }}_${{ matrix.exec_profile }}_${{ github.run_id }}_${{ github.job }}_results
          path: ${{ matrix.test_profile }}_${{ matrix.exec_profile }}_results
          overwrite: true

      - uses: actions/upload-artifact@v4
        if: always()
        name: Upload log
        with:
          name: nextflow_${{ matrix.test_profile }}_${{ matrix.exec_profile }}_${{ github.run_id }}_${{ github.job }}.log
          path: .nextflow.log
          overwrite: true
```
What about this one?

Comment (jpfeuffer) [reply .github/workflows/ci.yml:1]: github.job is just "test" for all runs.
Just remove everything you tried and add NXF_VER as another suffix. That is the last part that is still defining the matrix.

Comment (jpfeuffer) [reply .github/workflows/ci.yml:1]: No runid, no job. Just the NXF_VER. (and the profiles)

Comment (ypriverol) [reply .github/workflows/ci.yml:1]: New changes done. Check now.

**Comment by jpfeuffer** [comment]
Does anyone know why it does not find the .nextflow.log anymore? Do we log to a different directory? Did nextflow change the default location?

**Comment by ypriverol** [comment]
> Does anyone know why it does not find the .nextflow.log anymore? Do we log to a different directory? Did nextflow change the default location?

Well spot, let me find out.

**Comment by ypriverol** [comment]
> Does anyone know why it does not find the .nextflow.log anymore? Do we log to a different directory? Did nextflow change the default location?

Solved.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -81,7 +82,7 @@ jobs:
         # For example: adding multiple test runs with different parameters
         # Remember that you can parallelise this by using strategy.matrix
         run: |
-          nextflow run ${GITHUB_WORKSPACE} -profile $TEST_PROFILE,$EXEC_PROFILE --outdir ${TEST_PROFILE}_${EXEC_PROFILE}_results
+          nextflow -log pipeline.log run ${GITHUB_WORKSPACE} -profile $TEST_PROFILE,$EXEC_PROFILE --outdir ${TEST_PROFILE}_${EXEC_PROFILE}_results
```
Comment (jpfeuffer) [inline .github/workflows/ci.yml:85]: Why change the log filename? This just confused everyone.

Comment (ypriverol) [reply .github/workflows/ci.yml:85]: I do think the plugin, tht we run the pipeline, delete or do something with the .nextflow.log. I a previous run I check all the files created in the run and they were not .nextflow.log, check this https://github.com/bigbio/quantms/actions/runs/11211083164/job/31159229622

Comment (jpfeuffer) [reply .github/workflows/ci.yml:85]: But who added this line? And why? It is not needed and broke the additional CI log uploads that I implemented.

Comment (ypriverol) [reply .github/workflows/ci.yml:85]: > Of course there is no .nextflow.log anymore if you overwrite the default here.

Before writing the -log option it was not there. Check my CI/CD before adding the `-log`. I did try to `ls` all the files but the .nextflow.log was not there already.

Comment (jpfeuffer) [reply .github/workflows/ci.yml:85]: Which plugin do you mean?

Comment (ypriverol) [reply .github/workflows/ci.yml:85]: Sorry misunderstood. I don't know why `.nextflow.log` is not there. I did the `ls -R -a`, but the file is not there, my point is that in other nf-core pipelines, it is not there.

Comment (jpfeuffer) [reply .github/workflows/ci.yml:85]: I'm not sure if it is not there in other pipelines. They just don't try to upload it.
They are clearly saying it exists in the docs:
https://nf-co.re/docs/usage/troubleshooting/basics#categorize-the-type-of-error

**Comment by jpfeuffer** [comment]
The problem is that your `ls -la` does not show hidden files..
We would have found out, that the file is actually there and that the upload-action@v4 has a new parameter include-hidden-files.

**Review by jpfeuffer — APPROVED** [review]

**Pull request merged**